lib/validators/upload_validator.rb

```diff
@@ -50,38 +50,50 @@ def maximum_attachment_file_size(upload)
 
   private
 
-  def authorized_uploads(upload)
-    authorized_uploads = Set.new
+  def extensions_to_set(exts)
+    extensions = Set.new
 
-    extensions = upload.for_theme ? SiteSetting.theme_authorized_extensions : SiteSetting.authorized_extensions
-
-    extensions
+    exts
       .gsub(/[\s\.]+/, "")
       .downcase
       .split("|")
-      .each { |extension| authorized_uploads << extension unless extension.include?("*") }
+      .each { |extension| extensions << extension unless extension.include?("*") }
+
+    extensions
+  end
 
-    authorized_uploads
+  def authorized_extensions(upload)
+    extensions = upload.for_theme ? SiteSetting.theme_authorized_extensions : SiteSetting.authorized_extensions
+    extensions_to_set(extensions)
   end
 
   def authorized_images(upload)
-    authorized_uploads(upload) & FileHelper.images
+    authorized_extensions(upload) & FileHelper.images
   end
 
   def authorized_attachments(upload)
-    authorized_uploads(upload) - FileHelper.images
+    authorized_extensions(upload) - FileHelper.images
   end
 
   def authorizes_all_extensions?(upload)
+    if upload.user&.staff?
+      return true if SiteSetting.authorized_extensions_for_staff.include?("*")
+    end
     extensions = upload.for_theme ? SiteSetting.theme_authorized_extensions : SiteSetting.authorized_extensions
-    extensions.include?("*")
+    return extensions.include?("*")
```

Comment:
You don't need that `return` ;) Ruby always return the value of the last expression